### ISSUE TITLE
Physio to Dicom feature interferes with FIRMM on XA30A

### ISSUE THREAD

**Issue by Chenjennie** [CMRR-C2P/MB#306]
Hi, we just upgraded one of our scanners to XA30A-SP02 and have been testing our previous protocols on the new system. I followed the instructions here to open up forbidden options with the MBAdvancedsettings.ini file because we need to output 16-bit dicoms. Since our old protocols had Dicom physio option on, I tried to test this function on XA30A, but our FIRMM tablet stopped recognizing the incoming scan. Changing physio back to none allowed FIRMM to recognize the scans but reconstruction lagged severely - scanner has gotten through 10 min of the scan but only the first 12s of data have been reconstructed. At the end of the scan, all of the images reconstructed within seconds. In case it's relevant, the scan had multiband acceleration of 8 and we're using the 64 channel head coil. FIRMM is set to receive images through BOLD addin. This was not an issue before I created the MBAdvancedsettings.ini file. Is there some additional setting I need to use in order to restore fast reconstruction or are the forbidden protocol options somehow interfering with recon in the background and therefore should be turned off completely?

Thanks!
Jennie

**Comment by neurosurd** [comment]
I dont know a solution for the FIRMM issue but we've had the same reconstruction issue. Its the PhysioLog reconstruction that takes a while. The size of the file is also much larger than what it was on VE11C. We've had to keep patient scan open until all reconstruction is complete. If you cant wait, then you can run retrorecon for the affected sequences.

**Comment by neurosurd** [comment]
forgot to mention, Closing MR View&Go prior to starting the MB-EPI scan with physio acquisition helps a bit with the recon time.

**Comment by xiangruili** [comment]
Apart from the physio recording issue, is there any other issue with XA30A+R17?
We are planning to upgrade the PrismaFit to XA30A+R17, and wonder if there is any other consideration to delay it.
Thanks.

**Comment by neurosurd** [comment]
You should get the XA30A SP02 version from Siemens. 
@eauerbach Hi Ed, any solution/workaround to the physiological data DICOM file size issue? It takes a long time to reconstruct the images with physio ON. That's problematic especially with real time transfer for something like motion tracking.

**Comment by eauerbach** [comment]
I made a change to the way the physiology DICOMs are saved in the next version (R017pre6), which should be available shortly.

**Comment by neurosurd** [comment]
That's great news! Thanks!

**Comment by moonchgm** [comment]
Hello,

We tested new MB sequence, prep6 with saving DICOM file.  No issue during scan and recon found - PhysioLog series was successfully created. I tried to extract physiology signals from physio DICOM file (pulse, respiration) using readCMRRPhysio.m script, but found the names of dicomInfo field in XA30A file are different from old one. In addition, the dicomInfo structure is a little different between enhanced vs non-enhanced DICOM archiving even at same XA30A.  Here is comparison of dicominfo structure between VE11 vs. XA30A enhanced DICOM file, 

%{
VE11 DICOM phyiso sample,
...
                      AcquisitionNumber: 711
                         InstanceNumber: 1
              Private_0029_10xx_Creator: 'SIEMENS CSA NON-IMAGE'
              Private_0029_11xx_Creator: 'SIEMENS CSA HEADER'
              Private_0029_12xx_Creator: 'SIEMENS MEDCOM HEADER2'
                      Private_0029_1008: 'SPEC NUM 4'
                      Private_0029_1009: 'syngo MR E11'
                      Private_0029_1108: 'SPEC NUM 4'
                      Private_0029_1109: '20230103'
                      Private_0029_1110: [12552×1 uint8]
                      Private_0029_1118: 'MR'
                      Private_0029_1119: '20230103'
                      Private_0029_1120: [148768×1 uint8]
                      Private_0029_1260: 'com'
        PerformedProcedureStepStartDate: '20230103'
        PerformedProcedureStepStartTime: '143108.367000'
               PerformedProcedureStepID: 'MR20230103143108'
      PerformedProcedureStepDescription: 'Brain^res-uui'
    CommentsOnThePerformedProcedureStep: ''
              Private_7fe1_10xx_Creator: 'SIEMENS CSA NON-IMAGE'
                      Private_7fe1_1010: [728064×1 uint8]
%}
%{
XA30A enhanced DICOM physio example,

                           AcquisitionNumber: 737
                              InstanceNumber: 1
                         FrameOfReferenceUID: ''
                  PositionReferenceIndicator: ''
               DimensionOrganizationSequence: [1×1 struct]
                      DimensionIndexSequence: [1×1 struct]
                   Private_0021_12xx_Creator: 'SIEMENS MR SDR 01'
                   Private_0021_14xx_Creator: 'SIEMENS MR MRS 05'
                           Private_0021_1201: 'ACQUISITION'
                           Private_0021_1202: 'ACQUISITION'
                           Private_0021_1401: 343.1180
                           Private_0021_1405: 240
                              NumberOfFrames: 1
                                        Rows: 737
                                     Columns: 512
                               DataPointRows: 1
                            DataPointColumns: 0
                         SignalDomainColumns: 'TIME'
                          DataRepresentation: 'COMPLEX'
             PerformedProcedureStepStartDate: '20230501'
             PerformedProcedureStepStartTime: '204354.600000'
               PerformedProcedureStepEndDate: '20230501'
                    PerformedProcedureStepID: 'USd18bba76826d4a'
           PerformedProcedureStepDescription: ''
                  AcquisitionContextSequence: [1×1 struct]
            IssueDateOfImagingServiceRequest: '20230501'
              SharedFunctionalGroupsSequence: [1×1 struct]
            PerFrameFunctionalGroupsSequence: [1×1 struct]
                            SpectroscopyData: [188672×1 single]

%}

Here is physio DICOM files and CMRR BOLD protocol files attached. Is there any way to extract the physiology signal from current XA30A format?
[DICOMPhysio.zip](https://github.com/CMRR-C2P/MB/files/11401714/DICOMPhysio.zip)

Thanks for your help,

Chan

**Comment by brainnetworklab** [comment]
Hi Chan, Were you able to figure this out? I'm having the same issue.

**Comment by moonchgm** [comment]
Hi,

I don't have solution for physio DICOM. But I record XA30A physio log files of respiration and pulse during a BOLD scan ~10min by using "ideacmdtools", confirmed the data are correctly saved in the log files. Try the attached sample data and testing script. In XA30A log file, the tagging is a little different from VE11 - and this information could be helpful to decode physio DICOM assuming the data saved correctly. 

[PygioLog.zip](https://github.com/CMRR-C2P/MB/files/11499664/PygioLog.zip)

Chan

**Comment by eauerbach** [comment]
I updated the readCMRRPhysio.m to read XA data

**Comment by moonchgm** [comment]
I further tested physio recording in DICOM at XA30A, it seems there is no data except log information which is read by new readCMRRPhysio.m. In the physio DICOM file, there should be respiration and pulse data. This happened in other record of physio DICOM before. Is there any chance to fix the issue in MB EPI BOLD sequence? See the attached physio DICOm file.
[CMRR_Physio_Tes.MR._.8.1.2023.05.24.20.05.49.477.18554674.dcm.zip](https://github.com/CMRR-C2P/MB/files/11560851/CMRR_Physio_Tes.MR._.8.1.2023.05.24.20.05.49.477.18554674.dcm.zip)

**Comment by eauerbach** [comment]
There will only be log data present in the DICOM file if it is significant. For example, if you are scanning a phantom and there are no actual meaningful respiration and pulse signals detected, no log data will be saved.

**Comment by moonchgm** [comment]
The physio DICOM was acquired while BOLD scanned from a human subject. Separately, physio log files were acquired during fMRI scan from same subject/session. The physio log data showed respiratory and pulse data, but no resp/pulse data in physio DICOM. Sample data upload. Any solution for physio DICOM in MB BOLD?
[XA30APhysio.zip](https://github.com/CMRR-C2P/MB/files/11779232/XA30APhysio.zip)

**Comment by brainnetworklab** [comment]
Hi again Chan. You were able to get the BOLD to work without recon issues. What did you change? Still running into the recon issues. We're running pre6 on XA30.

**Comment by moonchgm** [comment]
Hello,

No BOLD recon issues are found, except physio DICOM (no physio data saved). I run latest BOLD sequence for MB fMRI.

Chan

From: brainnetworklab ***@***.***>
Date: Friday, July 21, 2023 at 7:01 PM
To: CMRR-C2P/MB ***@***.***>
Cc: moonchgm ***@***.***>, Comment ***@***.***>
Subject: Re: [CMRR-C2P/MB] Physio to Dicom feature interferes with FIRMM on XA30A (Issue #306)

Hi again Chan. You were able to get the BOLD to work without recon issues. What did you change? Still running into the recon issues. We're running pre6 on XA30.

**Comment by eauerbach** [comment]
I will make pre7 available soon with some changes to the physio dicom format. Hopefully it will be better. It is still a mystery why this is happening.

**Comment by moonchgm** [comment]
Great! I can test the functioning as soon as possible. Thanks. Chan

From: Eddie Auerbach ***@***.***>
Date: Friday, August 11, 2023 at 1:07 PM
To: CMRR-C2P/MB ***@***.***>
Cc: moonchgm ***@***.***>, Comment ***@***.***>
Subject: Re: [CMRR-C2P/MB] Physio to Dicom feature interferes with FIRMM on XA30A (Issue #306)

I will make pre7 available soon with some changes to the physio dicom format. Hopefully it will be better. It is still a mystery why this is happening.

**Comment by dfabbott** [comment]
At our site using R017pre6 XA50 on Vida, when any option other that physio logging “off” is chosen, the sequences won’t run, and the scanner gives an “image reconstruction failed” error.
@eauerbach, I can send to you a text file with errors copied from the log viewer on the scanner in case that helps diagnose the problem.

**Comment by eauerbach** [comment]
Thanks, now I see that Siemens changed the format of the physiological data starting with XA31, which explains why XA31 and XA50 are stopping with an error.

Separately, I think I have found that the delay in processing the physiology DICOM images in XA30 is due to some additional ICE processing Siemens is adding automatically, which isn't needed. I assume this also affects XA20/31/50.

I will address both of these issues and release pre7 shortly. I may eventually need some raw data from a XA31/50 system if there are lingering issues around the new data format.

**Comment by peter-mri** [comment]
> I will make pre7 available soon with some changes to the physio dicom format. Hopefully it will be better. It is still a mystery why this is happening.

Hello, is there any estimated timeline for release of pre7? We have recently "upgraded" our PRISMA to XA 30 (SP02) and are using R017 pre6. The recording of physio signals seems to be an issue, partially as described already (recording to DICOM extends the recon time by about the same time as acquisition). However, I've also done a fairly basic test with PMU simulation, when I turned off the simulated physio signals half way through acquisition, but I didn't see that reflected in the signal saved in DICOM, which makes me question whether the physio recording is happening after the image acquisition, or with some kind of shift? I can share my phantom/simulated data if it helps.

**Comment by eauerbach** [comment]
I am building pre7 for the XA versions now, it will be available in the next day or so. The reconstruction delay seen on XA30 should be fixed, and it should no longer crash when starting a scan on XA31 and XA50.

**Comment by neurosurd** [comment]
@eauerbach Just tested the pre7 on XA30 and the dicom physio+ recon  is working much faster. Minor bug in the new version is that the triggering scheme under "Physio" sequence card is at EXT/TRIG by default. So we have to manually change it back to none.

**Comment by DanielGuellmar** [comment]
@eauerbach I've also installed and tested the pre7 Version on XA30 (Prisma). Simultaneous physio recording by the sequence is now working without delay (64-channel-coil, pulse, and resp). Before pre7 installation, we observed that after the initial sequence phase, only three objects/volumes were reconstructed, and the recon waited until the end of each scan before it continued to reconstruct the "rest" of the data. 

I also can confirm the issue raised by @neurosurd. All existing CMRR SMS bold protocols, which had within-sequence physio recording enabled, were modified to wait for "Ext/Physio" trigger in the "Physio" task card upon opening the protocol after pre7 installation. I would also consider this a minor bug, but is it/was part of the solution?

A big thank you to @eauerbach for solving this problem.

**Comment by rwmair** [comment]
> I also can confirm the issue raised by @neurosurd. All existing CMRR SMS bold protocols, which had within-sequence physio recording enabled, were modified to wait for "Ext/Physio" trigger in the "Physio" task card upon opening the protocol after pre7 installation. I would also consider this a minor bug, but is it/was part of the solution?

I was just working yesterday on a Prisma newly transitioned from VE11C to XA30-SP02. I had the same exact issue - however not just CMRR SMS-BOLD but also CMRR SMS-diffusion scans as well.  Every one had to have 1stSignal/Mode on the Physio tab changed from "Ext Trigger" back to "None". This applied to protocols converted from the existing VE11C protocol tree, as well as those previously created on a different XA30 scanner and imported via an exar1 file. The exar1 file import gave no indication that there was a conversion error/note, or that anything had changed during the import.

**Comment by eauerbach** [comment]
Yes, sorry, this is an unintended side effect of the fix for XA that I did not catch. If you have existing protocols or are converting protocols from a VD/VE version where the DICOM physiology option was previously enabled, an unexpected Ext/Physio trigger on the Physio task card will appear and it will be necessary to disable it manually (one time only). For newly created protocols it is not a problem. I will correct this in a future update.

**Comment by eauerbach** [comment]
Can you try the latest version on the web site (dated 24-Oct-2023). Hopefully this fixes the issue with existing/converted protocols. (Currently the update is only posted for XA30)

**Comment by mharms** [comment]
The "24-Oct-2023" version appears to resolve this issue.

- We had installed the "3-Oct-2023" version of "pre7" a couple days ago on our XA30 Prisma, and I confirmed that scans with "Physio recording" set to DICOM (on the Sequence:Special tab) had 1stSignal/Mode on the Physio tab changed to "Ext./Trigger".
- After installing the "24-Oct-2023" version this morning, those same scans had 1stSignal/Mode back at None.
- Confirmed that a fresh import of a previously generated XA30 exar1 with scans with "Physio recording" = DICOM still had 1stSignal/Mode = None after import.
- Recon of a BOLD with "Physio recording" = DICOM was ongoing during the scan, as seen in inline viewer.
- Series and its associated PhysioLog file showed up in Patient Browser within a minute of completion of scan. (Perhaps even sooner, but a minute after scan ended is when I checked).
